akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -590,16 +590,18 @@ private[akka] class ShardRegion(
 
   def receiveCommand(cmd: ShardRegionCommand): Unit = cmd match {
     case Retry ⇒
+      sendGracefulShutdownToCoordinator()
+
       if (shardBuffers.nonEmpty)
         retryCount += 1
       if (coordinator.isEmpty)
         register()
       else {
-        sendGracefulShutdownToCoordinator()
         requestShardBufferHomes()
-        tryCompleteGracefulShutdown()
       }
 
+      tryCompleteGracefulShutdown()
+
```

Comment:
Was a bit confused first before I realised that `sendGracefulShutdownToCoordinator` only actually does that if there is a graceful shutdown in progress, I think the name is a bit misleading, but I can't think of anything making it more clear otoh.